al-por-menor/front-end/components/Profile.tsx

```diff
@@ -0,0 +1,87 @@
+"use client";
+import { Frequency } from "@/enums/enums";
+import { ISubscription } from "@/types/responses";
+import React, { useState } from "react";
+import SubscriptionComponent from "./SubscriptionComponent";
+
+// Interface for user
+interface User {
+  id: number;
+  name: string;
+  email: string;
+  subscriptions: ISubscription[];
+}
+
+// Profile component
+const ProfilePage: React.FC = () => {
+  // User state
+  const [user, setUser] = useState<User>({
+    id: 1,
+    name: "John Doe",
+    email: "johndoe@example.com",
+    subscriptions: [
+      {
+        subscriptionId: "1",
+        imgUrl: "subscription_image1.jpg",
+        productName: "Pack 1",
+        frequency: Frequency.MONTHLY,
+        quantity: 2,
+        price: 10,
+      },
+      {
+        subscriptionId: "2",
+        imgUrl: "subscription_image2.jpg",
+        productName: "Pack 2",
+        frequency: Frequency.WEEKLY,
+        quantity: 1,
+        price: 5,
+      },
+    ],
+  });
```

Comment:
esto cambió ligeramente x el back. te paso el correcto:
```suggestion
const ProfilePage: React.FC = () => {
  // User state
  const [user, setUser] = useState<User>({
    id: 1,
    name: "John Doe",
    lastName: "",
    email: "johndoe@example.com",
    address: "",
    city: "",
    password: "",
    supplier: false,
    otpCode: 0,
    otpExpiration: new Date(),
    subscriptions: [
      {
        userId: 1,
        subscriptionId: "1",
        cadence: "MONTHLY",
        numberOfPeople: 1,
      },
      {
        userId: 1,
        subscriptionId: "2",
        cadence: "WEEKLY",
        numberOfPeople: 1,
      },
    ],
  });
```

Reply:
Se podria agregar el nombre del pack, su precio e imagen en este response? Asi las mostramos en el FE

Reply:
yes. el ´name´ e ´imgUrl´ estan en el modelo "subscription". el precio aún no preparé el endpoint pero mandale price y fue.